source/helpers/transport-json.js

```diff
@@ -23,30 +23,56 @@ export default class TransportJSON extends Transport {
       this.emit('logged', log);
     });
     const sanitizedLogMessage = stripAnsi(log.message.replace('\n', ' ').trim());
-    if (log.level === 'info' || log.level === 'warn') {
-      if (log.isMessage) this.state.message += sanitizedLogMessage;
-      else this.state.logs.push(sanitizedLogMessage);
-    } else if (log.level === 'debug') {
-      this.state.debugModeLogs.push(`${log.callerFunction}: ${sanitizedLogMessage} ${log.ms}`);
-    } else if (log.level === 'error' && log.errorCode) {
-      if (!this.state.status !== 'error') {
+    switch (true) {
+      case log.level === 'info' || log.level === 'warn':
+        if (log.isMessage) this.state.message += sanitizedLogMessage;
+        else this.state.logs.push(sanitizedLogMessage);
+        break;
+      case (log.level === 'debug'):
+        this.state.debugModeLogs.push(`${log.callerFunction}: ${sanitizedLogMessage} ${log.ms}`);
+        break;
+      case (log.level === 'error' && log.errorCode !== undefined):
         this.state.status = 'error';
-      }
-      if (!this.state.code) this.state.code = log.errorCode;
-      this.state.errors.push({
-        code: log.errorCode,
-        field: log.field ? log.field : null,
-        message: sanitizedLogMessage,
-      });
-    } else {
-      this.state.code = ERROR_CODES.EXCEPTION_001;
-      this.state.status = 'error';
-      this.state.errors.push({
-        code: ERROR_CODES.EXCEPTION_001,
-        field: null,
-        message: sanitizedLogMessage,
-      });
+        this.state.code = log.errorCode;
+        this.state.errors.push({
+          code: log.errorCode,
+          field: log.field ? log.field : null,
```

Comment:
If you want to leverage the optional chaining syntax, then use `log?.field ?? null`

Reply:
I would be using `log.field || null` for shorter syntax